cwe_checker_rs/src/intermediate_representation/expression.rs

```diff
@@ -76,6 +76,62 @@ impl Expression {
             Cast { size, .. } | Unknown { size, .. } | Subpiece { size, .. } => *size,
         }
     }
+
+    /// Substitute some trivial expressions with their result.
+    /// E.g. substitute `a XOR a` with zero or substitute `a OR a` with `a`.
+    pub fn substitute_trivial_operations(&mut self) {
+        use Expression::*;
+        match self {
+            Var(_) | Const(_) | Unknown { .. } => (),
+            Subpiece {
+                low_byte,
+                size,
+                arg,
+            } => {
+                arg.substitute_trivial_operations();
+                if *low_byte == ByteSize::new(0) && *size == arg.bytesize() {
```

Comment:
Just for my understanding. You'll simplify the subpiece operation when the size of the subpiece is 0? So you basically remove subpiece und just have the underlying expression?

Reply:
No, the size of the subpiece is the same as the size of the argument. And the subpiece starts at the zeroth byte of the argument (`low_byte = 0`). Or in other words, the subpiece is exactly the same as the argument.